pkg/controller/pod.go

```diff
@@ -202,15 +214,93 @@ func (c *Controller) syncPod(ctx context.Context, pod *corev1.Pod) error {
 		}
 	}
 
-	klog.V(4).Infof("Registering instance %s under service %+v", pod.Name, cloudmapConfig)
+	cloudMapServiceKey := *(awssdk.String(*cloudmapConfig.NamespaceName)) + "-" +
```

Comment:
why we have these double indirects here...i believe *(awssdk.String(*cloudmapConfig.NamespaceName)) == `*cloudmapConfig.NamespaceName`.
And it will be nil safe to just use `awssdk.StringValue(cloudmapConfig.NamespaceName)`

Reply:
Yes, Copy-Paste issue here, will correct it.